pkg/controller/linstornodeset/linstornodeset_controller.go

```diff
@@ -410,11 +407,6 @@ func (r *ReconcileLinstorNodeSet) reconcileSatNodes(pns *piraeusv1alpha1.Linstor
 		if out.err != nil {
 			compoundError = append(compoundError, out.err)
 		}
-
-		// Guard against empty statuses.
-		if out.sat == nil || out.sat.NodeName != "" {
-			pns.Status.SatelliteStatuses[out.sat.NodeName] = out.sat
```

Comment:
Looks like `out.sat` is no longer necessary, so `satelliteStatusOut` only needs to be a channel of errors.

I have the feeling that the whole `satelliteStatusIn`, `satelliteStatusOut` logic is unnecessarily complicated. Doing this concurrently with a concurrent operator limit is reasonable, but I don't see why it needs to be split into 2 stages. But that's a different issue, apart from the potential to remove `type satStat` now.